src/main/java/com/box/sdk/BoxSearchParameters.java

```diff
@@ -0,0 +1,427 @@
+package com.box.sdk;
+import java.util.Date;
+import java.util.List;
+/**
+ * Used to Setup Box Search Parameters
+ *
+ * <p>Advanced Search support here allows a number of parameters be specified to take full advantage of
+ * box search capabilities. Query parameter is required in all cases except when Metadata templates
+ * searching is being used.</p>
+ *
+ */
+public class BoxSearchParameters {
+    private String query = null;
```

Comment:
You can omit setting these fields to null and 0. They'll be automatically assigned those as default values.